test/com/google/javascript/jscomp/Es6VariableReferenceCheckTest.java

```diff
@@ -277,6 +277,10 @@ public void testClass() {
     assertNoWarning("class A { f() { return 1729; } }");
   }
 
+  public void testRedeclareClassName() {
+    assertNoWarning("var Clazz = class Foo {}; var Foo = 3;");
```

Comment:
I think we should support it in the compiler, but discourage it in the style guide.